ocaml/xenopsd/xc/domain.ml

```diff
@@ -27,6 +27,14 @@ open D
 
 let finally = Xapi_stdext_pervasives.Pervasiveext.finally
 
+module Compression = struct
+  let scheme = "gzip"
```

Comment:
`gzip` is quite slow for compression.
Would be useful to use something like `lz4` or `zstd -1` (there is already support for zstd in xapi-compress).
Lz4 is very fast (at the expense of slight reduction in compression ratio), and `zstd` is both faster and compresses better than gzip (depending on setting). `gzip` is only reasonably fast for decompression.
See benchmarks at https://lz4.github.io/lz4/ and https://facebook.github.io/zstd/#benchmarks

What matters here is compression time + network transfer time + decompression time. Although lets say compression and decompression runs on different machines and in parallel, so: `max(compression time, decompression time) + network transfer time`. Network transfer time will depend on compression ratio, so there is a tradeoff between time taken to compress a little bit more, or being a lot faster on compression and transferring a little bit more.
The lz4 benchmarks take into consideration the total transfer time: <2s for LZ4, vs >11s for zlib (gzip)

Reply:
Quoting from the ztd benchmarks:



Compressor name | Ratio | Compression | Decompress.
-- | -- | -- | --
zlib 1.2.11 -1 | 2.743 | 90 MB/s | 400 MB/s
zstd 1.4.5 -1 | 2.884 | 500 MB/s | 1660 MB/s
zstd 1.4.5 --fast=1 | 2.434 | 570 MB/s | 2200 MB/s
zstd 1.4.5 --fast=3 | 2.312 | 640 MB/s | 2300 MB/s
zstd 1.4.5 --fast=5 | 2.178 | 700 MB/s | 2420 MB/s
lz4 1.9.2 | 2.101 | 740 MB/s | 4530 MB/s
```